_posts/2019-06-25-lambda-quickstart.md

```diff
@@ -0,0 +1,320 @@
+When an OpenFaaS functions executes in AWS Lambda it needs an IAM role. This role will give functions access to other
+AWS services. Currently we only support one role that is assigned to all functions. In the future we will change this
+to be per function. 
+
+If your functions do not require access please follow the steps to create a [blank role](#232-no-access-function-iam-role)
+
+##### 2.3.1. Restricted function IAM Role
+Your requirements will be specific to you, in this example I am giving functions access to a particular s3 bucket:
+
+
+```javascript
+{
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Sid": "CreateLogs",
+            "Effect": "Allow",
+            "Action": [
+                "logs:CreateLogGroup",
+                "logs:CreateLogStream",
+                "logs:PutLogEvents"
+            ],
+            "Resource": "*"
+        },
+        {
+            "Sid": "ListObjectsInBucket",
+            "Effect": "Allow",
+            "Action": [
+                "s3:ListBucket"
+            ],
+            "Resource": [
+                "arn:aws:s3:::kubecon-2019",
+                "arn:aws:s3:::containerdays-2019"
+            ]
+        },
+        {
+            "Sid": "AllObjectActions",
+            "Effect": "Allow",
+            "Action": "s3:*Object",
+            "Resource": [
+                "arn:aws:s3:::kubecon-2019/*",
+                "arn:aws:s3:::containerdays-2019/*"
+            ]
+        }
+    ]
+}
+``` 
+
+- Create a [new IAM policy](https://docs.aws.amazon.com/IAM/latest/UserGuide/access_policies_create.html#access_policies_create-start), with the above minimum permissions. For example call this new policy `OpenFaaS Lambda Function Policy`.
+- Create a [new IAM role](https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles_create_for-user.html), and assign it the above police. For example call this new role `OpenFaaS Lambda Function Role`. 
+
+OR
+
+##### 2.3.2. No access function IAM Role
+
+```javascript
+{
+    "Version": "2012-10-17",
+    "Statement": [
+       {
+           "Sid": "CreateLogs",
+           "Effect": "Allow",
+           "Action": [
+               "logs:CreateLogGroup",
+               "logs:CreateLogStream",
+               "logs:PutLogEvents"
+           ],
+           "Resource": "*"
+       }
+    ]
+}
+``` 
+
+- Create a [new IAM policy](https://docs.aws.amazon.com/IAM/latest/UserGuide/access_policies_create.html#access_policies_create-start), with the above minimum permissions. For example call this new policy `OpenFaaS Lambda Function Policy`.
+- Create a [new IAM role](https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles_create_for-user.html), and assign it the above policy. For example call this new role `OpenFaaS Lambda Function Role`. 
+
+#### 2.4. Assign AWS access credentials to faas-lambda using Kubernetes secrets
+This is a secure way to pass the AWS access credentials to the `faas-lambda` provider
```

Comment:
`This is a secure way to pass the AWS access credentials`

How about:

`Create a secret to store the AWS credentials in the Kubernetes secret store?`